api/batch/controller.go

```diff
@@ -285,6 +285,9 @@ func (c *controller) syncStatus(ctx context.Context, key string) error {
 		c.cleanup(ctx, predictionJob, sparkApp.Namespace)
 		modelName := getModelName(predictionJob.Name)
 		BatchCounter.WithLabelValues(sparkApp.Namespace, modelName, string(predictionJob.Status)).Inc()
+		if predictionJob.Status == models.JobTerminated {
+			return c.store.Delete(predictionJob)
+		}
```

Comment:
Is this a new behaviour we're implementing? i.e. a job gets deleted automatically and immediately after it's terminated?

Reply:
yes, i was thinking the same implementation from turing, the runner will delete any job that in terminated state

Reply:
@krithika369 @ariefrahmansyah, there should be no concerns with this change right? The only reason we used to leave terminated jobs dangling in the DB was for debugging?

Reply:
Since i already change the implementation to [this](https://github.com/caraml-dev/merlin/pull/389#discussion_r1236503432), the runner now doesn't delete the terminated jobs, since the StopFunction will immediately Delete the jobs. But still, is it okay if i do this? wdyt?

Reply:
So if I understand correctly, these are the two workflows for stopping prediction jobs?
- runner: (if it operates on a 'terminated' job) -> updates db that the job is terminated 
- stop API endpoint: (if called) -> stop job -> delete job (changed)

Reply:
no there's only one way to delete the prediction job, from stop API endpoint
the runner will just update the prediction job status according the spark application, so the runner will only update the prediction job status, eg: pending -> running -> completed

Reply:
Oops my bad, I was thinking of two different things when writing the first sentence and when listing the bullet points, but yes I had my doubt clarified 😅

Reply:
@ariefrahmansyah do you see any concerns with this change (`api/batch/controller.go`) - immediately deleting the terminated jobs from DB / cluster?

Reply:
I have no concern with immediately deleting the terminated jobs. Technically, it shall be safe with no side effect.